### ISSUE TITLE
Rebuild packages fails in Beta

### ISSUE THREAD

**Issue by alex-ketch** [atom/atom#9580]
Similar to https://github.com/atom/atom/issues/8837, however this only happens with the Atom Beta build, no _Incompatible Package_ issues in stable build.

Upon opening Atom, even in safe mode, a red bug icon appears. The screen it leads to reveals that

> Some installed packages could not be loaded because they contain native modules that were compiled for earlier version of Atom.

Clicking the button to `Rebuild Packages` fails giving the errors shown below.
Note the target path for the Make command is odd and seems to point to the non-beta version.
`../../../../../../../../../../../../Applications/Atom`

![screen shot 2015-11-13 at 19 03 10](https://cloud.githubusercontent.com/assets/1646307/11155004/9e33161c-8a3a-11e5-90e0-41f047edeaeb.png)

If i navigate to the package folder and run the rebuild command, it succeeds with `apm` but fails with `ape-beta` giving same output as above.

<img width="713" alt="screen shot 2015-11-13 at 18 48 54" src="https://cloud.githubusercontent.com/assets/1646307/11155082/3d85eec4-8a3b-11e5-8c64-02e3ecbe223a.png">

This is on OS X 10.11.1, running NVM if it makes a difference.

**Comment by benogle** [comment]
cc @maxbrunsfeld @nathansobo

**Comment by benogle** [comment]
I wonder if this has something to do with the app naming in beta?

**Comment by joefitzgerald** [comment]
Also note the use of node `0.10.40` which is not expected.

**Comment by alex-ketch** [comment]
Don't know where `0.10.40` is coming from, the output of `nvm ls` and respective paths are below. 
``` shell
~|⇒ nvm ls   
        v0.12.7
->       v4.1.1
default -> stable (-> v4.1.1)
node -> stable (-> v4.1.1) (default)
stable -> 4.1 (-> v4.1.1) (default)
iojs -> N/A (default)

~|⇒ which node
/Users/Ketch/.nvm/versions/node/v4.1.1/bin/node

~|⇒ which npm
/Users/Ketch/.nvm/versions/node/v4.1.1/bin/npm
```

**Comment by pwnall** [comment]
I've also stumbled upon this bug while using atom-beta 1.12 on OSX 10.12.1. I also ran into this bug on OSX 10.11.6, so I don't think it depends on a specific version of OSX. I'm guessing the bug has to do with the app's name in beta.

```
gyp WARN download NVM_IOJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use IOJS_ORG_MIRROR
gyp WARN download NVM_IOJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use IOJS_ORG_MIRROR
gyp WARN download NVM_IOJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use IOJS_ORG_MIRROR
make: *** No rule to make target `../../../../../../../../Applications/Atom', needed by `Makefile'.  Stop.
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Applications/Atom Beta.app/Contents/Resources/app/apm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 16.1.0
gyp ERR! command "/Applications/Atom Beta.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom Beta.app/Contents/Resources/app/apm/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/costan/.atom/packages/git-blame/node_modules/pathwatcher
gyp ERR! node -v v4.4.5
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 

npm ERR! Darwin 16.1.0
npm ERR! argv "/Applications/Atom Beta.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom Beta.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/costan/.atom/.apm/.apmrc" "--userconfig" "/Users/costan/.atom/.apmrc" "rebuild" "--runtime=electron" "--target=1.3.6" "--arch=x64"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! pathwatcher@6.7.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pathwatcher@6.7.0 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pathwatcher package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs pathwatcher
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls pathwatcher
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/costan/.atom/packages/git-blame/npm-debug.log
```

**Comment by smashwilson** [comment]
This should be fixed once atom/apm#673 is fixed. :sparkles:
